drivers: sensor: icp201xx: fix channel check logic

kartben opened the pull request:
Update the channel validation logic to avoid using enum constants as boolean values.
Fixes clang warning "converting the enum constant to a boolean" observed in main https://github.com/zephyrproject-rtos/zephyr/actions/runs/14310267206/job/40103313166

kartben committed:
drivers: sensor: icp201xx: fix channel check logic

Update the channel validation logic to avoid using enum constants as
boolean values.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

pdgendt commented on drivers/sensor/tdk/icp201xx/icp201xx_drv.c:
```diff
@@ -216,8 +216,8 @@ static int icp201xx_channel_get(const struct device *dev, enum sensor_channel ch
 
-	if (!(chan == SENSOR_CHAN_AMBIENT_TEMP || chan == SENSOR_CHAN_PRESS ||
-	      SENSOR_CHAN_ALTITUDE)) {
+	if (chan != SENSOR_CHAN_AMBIENT_TEMP && chan != SENSOR_CHAN_PRESS &&
+	    chan != SENSOR_CHAN_ALTITUDE) {
 		return -ENOTSUP;
 	}
```
After this, the `chan` is checked again, maybe the body should just be changed to

```c
	if (chan == SENSOR_CHAN_PRESS) {
		icp201xx_mutex_lock(dev);
		icp201xx_convert_pressure(val, data->raw_pressure);
		icp201xx_mutex_unlock(dev);
	} else if (chan == SENSOR_CHAN_AMBIENT_TEMP) {
		icp201xx_mutex_lock(dev);
		icp201xx_convert_temperature(val, data->raw_temperature);
		icp201xx_mutex_unlock(dev);
#ifdef CONFIG_FPU
	} else if (chan == SENSOR_CHAN_ALTITUDE) {
		struct sensor_value pressure_val, temp_val;
		float pressure, temperature, altitude;

		icp201xx_mutex_lock(dev);
		icp201xx_convert_pressure(&pressure_val, data->raw_pressure);
		icp201xx_convert_temperature(&temp_val, data->raw_pressure);
		pressure = pressure_val.val1 + ((float)pressure_val.val2 / 1000000);
		temperature = temp_val.val1 + ((float)temp_val.val2 / 1000000);
		altitude = convertToHeight(pressure, temperature);
		sensor_value_from_float(val, altitude);
		icp201xx_mutex_unlock(dev);
#endif
	} else {
		return -ENOTSUP;
	}
	return 0;
``

kartben commented:
I agree with @pdgendt that this might be a better fix but will let others chime in. The "hot" aspect of the fix is relative as it's a clang failure which doesn't directly impact PRs so given we know of a fix either way, I'm fine giving it another day or so to line up what folks see as the best solution

kartben commented:
@rbuisson-invn thoughts?

rbuisson-invn commented:
> @rbuisson-invn thoughts?
To be honest I do not even understand the issue here, nor the fix :)
I do not see any conversion from enum to boolean in the expression :

/__w/zephyr/zephyr/drivers/sensor/tdk/icp201xx/icp201xx_drv.c:219:70: error: converting the enum constant to a boolean [-Werror,-Wint-in-bool-context]
        if (!(chan == SENSOR_CHAN_AMBIENT_TEMP || chan == SENSOR_CHAN_PRESS ||
                                                                                                                                         ^

== operator should have precedence over || operator.
Sorry but, I would be glad if you could clarify this for me.

kartben committed:
drivers: sensor: icp201xx: fix channel check logic

Update the channel validation logic to properly check channel against
SENSOR_CHAN_ALTITUDE.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben commented:
@rbuisson-invn the latest push should make the fix more obvious - sorry for causing confusion by inverting the logic in my earlier fix :)

rbuisson-invn commented:
> @rbuisson-invn the latest push should make the fix more obvious - sorry for causing confusion by inverting the logic in my earlier fix :)

Far more clearer now, thanks !
